### ISSUE TITLE
oss-fuzz: Fix broken build

### ISSUE THREAD

**Issue by bshastry** [obgm/libcoap#231]
Hi,

The way oss-fuzz integration works now is that test harnesses are in this repo but the build script is the oss-fuzz repo [here](https://github.com/google/oss-fuzz/blob/master/projects/libcoap/build.sh)

Problem I noticed is the include line in the fuzz target [here](https://github.com/obgm/libcoap/blob/develop/tests/oss-fuzz/pdu_parse_target.c#L1) was changed in such a way that oss-fuzz build will fail. Has the Google bot complained about failed build? I am curious :smile: 

Since I also notice that @obgm has a commit that adds a make target for oss-fuzz tests, the best way to fix this issue would be for me to adapt the oss-fuzz build script to simply invoke something like `make -C test/oss-fuzz -f Makefile.ci check clean`

If you agree, I can go ahead with this approach.

**Comment by obgm** [comment]
Sorry, but the oss-fuzz build script is broken, see [line 23](https://github.com/google/oss-fuzz/blob/6b761f3826e6daecc6cf9cfcc737a1f8f1551c65/projects/libcoap/build.sh#L23). The correct flags as reported by pkg-config point to the sub-directory "include".

The change in pdu_parse_target.c was made to ensure that continuous integration does not fail.

@bshastry Can you elaborate on what you are proposing to change?

**Comment by bshastry** [comment]
I see. Okay, here's what I'll do: fix the include path in the oss-fuzz build script for libcoap. In the long run however, it's better to add a make target for oss-fuzz tests within libcoap repo. I notice that one such makefile exists but it doesn't accept env variables for custom compiler and linker flags: CC, CFLAGS, LDFLAGS etc.

**Comment by obgm** [comment]
Well, yes, it does:

```shell
$ make -C tests/oss-fuzz -f Makefile.ci CC=clang-6.0 CPPFLAGS=-I../../include CFLAGS="-x c" 
make: Entering directory '/home/bergmann/develop/libcoap/tests/oss-fuzz'
clang-6.0 -x c -I../../include  -c -o pdu_parse_target.o pdu_parse_target.c
```

**Comment by obgm** [comment]
The cleaner solution would be this:

```
./autogen.sh && ./configure --disable-doxygen --disable-manpages \
    && make -j$(nproc)
# This compiles and links fuzzer targets as is currently done by the
# oss-fuzz build script
make -C tests/oss-fuzz -f Makefile.oss-fuzz
```

**Comment by obgm** [comment]
See proposal in google/oss-fuzz#1676.

**Comment by bshastry** [comment]
There's a bug in the new proposal. To reproduce, do the following

```
$ git clone <oss-fuzz> && cd oss-fuzz
$ export PROJECT_NAME=libcoap
$ python infra/helper.py build_fuzzers --sanitizer address $PROJECT_NAME
...
make[1]: Entering directory '/src/libcoap/tests/oss-fuzz'
clang++ -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ -I../../include -I../../include   -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ /out/pdu_parse_target.o ../../.libs/libcoap-2.a    -o /out/pdu_parse_fuzzer
/usr/lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu/crt1.o: In function `_start':
(.text+0x20): undefined reference to `main'
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile.ci:26: recipe for target '/out/pdu_parse_fuzzer' failed
make[1]: *** [/out/pdu_parse_fuzzer] Error 1
make[1]: Leaving directory '/src/libcoap/tests/oss-fuzz'
Makefile.oss-fuzz:11: recipe for target 'oss-fuzz' failed
make: *** [oss-fuzz] Error 2
make: Leaving directory '/src/libcoap/tests/oss-fuzz'
fuzzers build failed.
```

Afaiu, LIBS is not getting passed through to the link step.

**Comment by bshastry** [comment]
Fixed the makefile in pending PR that is tagged with #218

**Comment by obgm** [comment]
Closed by commit 4980971.
